### PR TITLE
OGM-1592: Add 5.4.2.Final migration guide

### PR DESCRIPTION
**Pull request by jyemin**
https://hibernate.atlassian.net/browse/OGM-1592

### CODE REVIEW

File: documentation/manual/src/main/asciidoc/migration/index.asciidoc [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+end of life in April 2021).
+
+=== MongoDB Server
+
+Recent versions of the MongoDB server have removed support for some features on which hibernate-ogm depend. In particular:
+
+* MongoDB 4.2 removed support for the `$eval` command, on which hibernate-ogm support for stored procedures depends.
+
+All other hibernate-ogm tests of MongoDB support are passing with MongoDB 3.6, up through the latest release of MongoDB
+(currently MongoDB 8.0).
```
Comment (DavideD) [inline documentation/manual/src/main/asciidoc/migration/index.asciidoc:22]: On CI we only test up to 7, but I can upgrade it. Have you tried it?

File: documentation/manual/src/main/asciidoc/migration/index.asciidoc [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+with more recent versions of MongoDB that prevent 5.4.1.Final from executing native queries.
+
+As a consequence of the driver update, the oldest MongoDB server release that hibernate-ogm will connect to is now 3.6 (which reached
+end of life in April 2021).
+
+=== MongoDB Server
+
+Recent versions of the MongoDB server have removed support for some features on which hibernate-ogm depend. In particular:
+
+* MongoDB 4.2 removed support for the `$eval` command, on which hibernate-ogm support for stored procedures depends.
```
Comment (DavideD) [inline documentation/manual/src/main/asciidoc/migration/index.asciidoc:19]: It probably worth mentioning that native queries won't support `maxscan` and `snapshot` anymore: https://github.com/hibernate/hibernate-ogm/pull/1134/files#diff-8b6ff949d6524f98fc432f5eae5b7cec1c828e9ef1281930933b72ef0cc9cdb9L1241

**Comment by DavideD** [comment]
Merged, thanks.
I've also added a link to the existing migration guide: https://github.com/hibernate/hibernate-ogm/commit/133f0898cc86daa7e2c2dff3cda40facd0d7f20a

**Comment by yrodiere** [comment]
Hey @DavideD , did you check that this migration guide will actually get rendered?

**Comment by DavideD** [comment]
I didn't :/

Anyway, I will fix it in a bit

**Pull request merged**